sympy/solvers/solveset.py

```diff
@@ -1123,6 +1152,11 @@ def _solveset(f, symbol, domain, _check=False):
             result = FiniteSet(*[s for s in result
                       if isinstance(s, RootOf)
                       or domain_check(fx, symbol, s)])
+        elif isinstance(result,list):
```

Comment:
I think result here should be a Set. If it can be a list here then something has gone wrong somewhere

Reply:
Actually, this code was supplemented along with `solve` function(since `solve` return in form of a list...
yes, this should also be removed.

Reply:
This is due to the helper function from `_solve_as_bivariate ` (Line 1809)